packages/hash/api/src/lib/schemas/jsonSchema.ts

```diff
@@ -196,7 +194,10 @@ export class JsonSchemaCompiler {
     const schema = {
       ...partialSchema,
       $schema: jsonSchemaVersion,
-      $id: generateSchema$id(accountId, entityTypeId),
+      // The schema $id starts out by being the title.
+      // When the accountId and entityId of the schema is known, this can be replaced.
+      // We will keep this as a placeholder for any validation.
+      $id: title,
```

Comment:
Can we make it an optional argument? I realise we won't always be able to know the `accountId` and `entityId` when this is called, but when we do it'd be nice to just be able to pass the `$id` directly.

Reply:
Good idea. Should we still make the title the default value in case `$id` is not set?

Reply:
If `$id` is required by the scheme validator, I see no other way - yeah

Reply:
Seems that it is actually optional. I'll add it as an optional argument!

Reply:
Ah even better, in that case I think we can avoid setting `$id` to `title` entirely 👍